sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -175,12 +188,23 @@ private void setTaskStatus(Protos.TaskID taskID, Status status) {
             tasks.replace(taskID, new TaskStatusPair(tasks.get(taskID).getTaskInfo(), status));
             logger.info("Status for: {} is: {}", taskID.getValue(), status);
         }
+
+        Status newStatus = getStatus();
+        if (newStatus.equals(Status.PENDING)) {
```

Comment:
setting prepared before it's used within `getStatus(tasks)` is a bit sneaky. maybe pass in newStatus and have getStatus use that internally?